File: src/SOP8/Egovframework/Base/base-history-ibll/src/main/java/egovframework/base/history/ibll/IProcessManager.java
package egovframework.base.history.ibll;

import egovframework.base.history.ibll.request.*;
import egovframework.base.history.ibll.response.*;
import egovframework.base.excelreport.models.ResponseExcelInfo;

public interface IProcessManager {

    ResponseSOPHistory getSOPHistories(RequestSOPHistory data);

    ResponseSOPComponentHistory getSOPComponentHistories(RequestSOPComponentHistory data);

    ResponseExcelInfo downloadExcelSOPPartialHistory(RequestExcelSOPPartialHistory data);

    ResponseExcelInfo downloadExcelSOPAllHistory(RequestSOPHistory data);

    ResponseSensorDetectHistory requestSensorDetectHistory(RequestSensorDetectHistory data);

    ResponseSensorAnalysisHistory requestSensorAnalysisHistory(RequestSensorAnalysisHistory data);

    ResponseExcelInfo downloadExcelPartialSensorDetectHistory(RequestExcelPartialDetectHistory data);

    ResponseExcelInfo downloadExcelAllSensorDetectHistory(RequestExcelDetectHistory data);

    ResponseExcelInfo downloadExcelPartialSensorAnalysisHistory(RequestExcelPartialAnalysisHistory data);

    ResponseExcelInfo downloadExcelAllSensorAnalysisHistory(RequestExcelAnalysisHistory data);
}

File: src/SOP8/Base/History/Base.History.IBLL/IProcessManager.cs
using dnsExcelReport.Models;

namespace Base.History.IBLL
{
    using Request;
    using Response;

    public interface IProcessManager
    {
        ResponseSOPHistory GetSOPHistories(RequestSOPHistory data);
        ResponseSOPComponentHistory GetSOPComponentHistories(RequestSOPComponentHistory data);
        ResponseExcelInfo DownloadExcelSOPPartialHistory(RequestExcelSOPPartialHistory data);
        ResponseExcelInfo DownloadExcelSOPAllHistory(RequestSOPHistory data);
        ResponseSensorDetectHistory RequestSensorDetectHistory(RequestSensorDetectHistory data);
        ResponseSensorAnalysisHistory RequestSensorAnalysisHistory(RequestSensorAnalysisHistory data);
        ResponseExcelInfo DownloadExcelPartialSensorDetectHistory(RequestExcelPartialDetectHistory data);
        ResponseExcelInfo DownloadExcelAllSensorDetectHistory(RequestExcelDetectHistory data);
        ResponseExcelInfo DownloadExcelPartialSensorAnalysisHistory(RequestExcelPartialAnalysisHistory data);
        ResponseExcelInfo DownloadExcelAllSensorAnalysisHistory(RequestExcelAnalysisHistory data);
    }
}
